### PR TITLE
Add Guardian Light to members-data-api

### PR DESCRIPTION
**Pull request by david-pepper**
## Introduction
Add Guardian Light product to the /me and /mma endpoints in members-data-api. 

### Example responses

**/me**
```
{
  "userId": "200353814",
  "digitalSubscriptionExpiryDate": "2999-01-01",
  "guardianLightExpiryDate": "2025-12-10",
  "showSupportMessaging": false,
  "contentAccess": {
    "member": false,
    "paidMember": false,
    "recurringContributor": false,
    "supporterPlus": false,
    "feast": true,
    "digitalPack": true,
    "paperSubscriber": false,
    "guardianWeeklySubscriber": false,
    "guardianPatron": false,
    "guardianLight": true
  }
}
```

/me/mma
```
{
  "user": {
    "firstName": "David",
    "lastName": "Pepper",
    "email": "david.pepper+10122024_b@guardian.co.uk"
  },
  "products": [
    {
      "mmaCategory": "subscriptions",
      "tier": "Guardian Light",
      "isPaidTier": true,
      "selfServiceCancellation": {
        "isAllowed": true,
        "shouldDisplayEmail": true,
        "phoneRegionsToDisplay": [
          "UK & ROW",
          "US",
          "AUS"
        ]
      },
      "billingCountry": "United Kingdom",
      "joinDate": "2024-12-10",
      "optIn": true,
      "subscription": {
        "paymentMethod": "Card",
        "card": {
          "last4": "4242",
          "expiry": {
            "month": 1,
            "year": 2027
          },
          "type": "Visa",
          "stripePublicKeyForUpdate": "pk_test_Qm3CGRdrV4WfGYCpm0sftR0f",
          "email": "david.pepper+10122024_b@guardian.co.uk"
        },
        "contactId": "003UD00000LtB8QYAV",
        "deliveryAddress": {

        },
        "safeToUpdatePaymentMethod": true,
        "start": "2024-12-25",
        "end": "2025-12-10",
        "nextPaymentPrice": 200,
        "nextPaymentDate": "2024-12-25",
        "potentialCancellationDate": "2024-12-25",
        "lastPaymentDate": null,
        "chargedThroughDate": null,
        "renewalDate": "2025-12-10",
        "anniversaryDate": "2025-12-25",
        "cancelledAt": false,
        "subscriptionId": "A-S00943727",
        "trialLength": 13,
        "autoRenew": true,
        "plan": {
          "name": "Guardian Light",
          "price": 200,
          "currency": "£",
          "currencyISO": "GBP",
          "billingPeriod": "month"
        },
        "currentPlans": [],
        "futurePlans": [
          {
            "name": null,
            "start": "2024-12-25",
            "end": "2025-12-10",
            "shouldBeVisible": true,
            "chargedThrough": null,
            "price": 200,
            "currency": "£",
            "currencyISO": "GBP",
            "billingPeriod": "month",
            "features": ""
          }
        ],
        "readerType": "Direct",
        "accountId": "8ad09e54939569d10193b0556af741ee",
        "cancellationEffectiveDate": null
      }
    }
  ]
}
```

## Testing
- Unit tests have been added/updated to account for the new product
- Functional testing has taken place in the CODE environment, where the responses above were received

## Outstanding Questions
- Should we show support messaging for Guardian Light? 
 -- _It was decided that we would set this to true_
- Should the mma category be 'subscriptions' or 'recurringProduct'?
 -- _It was decided that we would set this to 'subscriptions'_

### CODE REVIEW

**Review by rupertbates — APPROVED** [review]
Looks good, just one small question 👍

File: membership-attribute-service/test/controllers/AttributeControllerTest.scala [hunk 1]
```diff
@@ -584,13 +609,44 @@ class AttributeControllerTest extends Specification with AfterAll with Idiomatic
...
-             |    "guardianPatron": false
+             |    "guardianPatron": false,
+             |    "guardianLight":false
              |  }
              |}""".stripMargin)
       verifyIdentityHeadersSet(result, userWithTierThreeId)
 
     }
 
+    "return the correct Guardian Light attributes for Guardian Light subscribers" in {
```
Comment (AndreaDiotallevi) [inline membership-attribute-service/test/controllers/AttributeControllerTest.scala:620]: New one!

**Review by AndreaDiotallevi — APPROVED** [review]
Nice work - looking good

File: membership-attribute-service/app/models/AccountDetails.scala [hunk 2]
```diff
@@ -217,6 +217,7 @@ object AccountDetails {
     case Product.Digipack => "subscriptions"
     case Product.SupporterPlus => "recurringSupport"
     case Product.TierThree => "recurringSupport"
+    case Product.GuardianLight => "subscriptions"
```
[the remark this answers is not in the document]
Comment (david-pepper) [reply membership-attribute-service/app/models/AccountDetails.scala:220]: Answered in outstanding questions (keeping it as 'subscriptions')

**Pull request merged**